### PR TITLE
Add color and vehicle type to VeRI chips

### LINKED ISSUE #65: Add color and vehicle type to VeRI chips
VeRI has color and vehicle type, so put it in the chips under the `misc` dictionary so we can use it to train.

### PR DESCRIPTION
**Pull request by agude**
This fixes #65.

### CODE REVIEW

**Comment by agude** [comment]
@d-grossman: I've added in two new values to the `misc` dictionary in VeRI chips. They are set to strings (https://github.com/Lab41/pelops/pull/66/files#diff-e20cb6b674d6b9fac3c868dac28b4504R57) for the Training set (the only place where we have the information) and to `None` for the test set chips.

Will this information suffice for training? Do you want different/other keywords?

File: pelops/datasets/veri.py [hunk 1]
```diff
@@ -34,31 +35,65 @@ class VeriDataset(chip.ChipDataset):
...
+
+            if vehicle_id is not None:
+                color = child.attrib.get("colorID", None)
+                vehicle = child.attrib.get("typeID", None)
+
+                try:
+                    color_id = int(color)
+                    vehicle_id = int(vehicle)
+                except ValueError:
+                    continue
```
Comment (d-grossman) [inline pelops/datasets/veri.py]: should let ppl know that there was an exception..
except ValueError as ex:
...         print('color:{0} or vehicle:{1} could not be converted to int'.format(color,vehicle))

Is there a logger for the project?

File: pelops/datasets/veri.py [hunk 2]
```diff
@@ -86,10 +121,14 @@ def __create_chip(self, img_dir, img_name):
...
+        filepath = os.path.join(img_dir, img_name)
         car_id = int(splitter[0])
         cam_id = int(utils.get_numeric(splitter[1]))
         time = datetime.datetime.fromtimestamp(int(splitter[2]))
         misc["binary"] = int(os.path.splitext(splitter[3])[0])
 
+        color, vehicle_type = self.__color_type.get(car_id, (None, None))
+        misc["color"] = color
+        misc["vehicle_type"] = vehicle_type
+
```
Comment (d-grossman) [inline pelops/datasets/veri.py]: should we in general be raising errors or making ppl look to see if a complex datatype returned from a function is filled in? (color, vehicle_type)
raise KeyError('car_id not found')

Comment (agude) [inline pelops/datasets/veri.py]: There are vehicles without color and type, mainly the entire test set. Give that that is the case, I think it is reasonable to fill in `None` as a default and let the user check.

File: pelops/datasets/veri.py (hunk 1 above)
Comment (agude) [inline pelops/datasets/veri.py]: I have addressed this problem as follows: I have removed all the `try`/`except` code, and replaced the `.get()` calls with pure dictionary access calls. This means that if a bogus ID or value is passed it, Python will throw a standard error and the stack trace will point you to the right place. This will let the user fix the issue.

I have run the new code on VeRI both test and training sets, and it parses fine, so the "standard" VeRI data doesn't have any edge cases here, any edge case is from the user modifying the files.

**Comment by agude** [comment]
I just rebased onto the current master so I can run tests in Docker.

**Comment by agude** [comment]
It builds and tests in docker now! Very nice!

**Comment by agude** [comment]
I have added a "more helpful" error for when the `xml` file is malformed:

```
ValueError: multi-byte encodings are not supported

The label file 'train_label.xml' comes malformed from the
source. The first line needs to be changed to:
'<?xml version="1.0" encoding="UTF-8" ?>'
if it is not already.
See: https://github.com/Lab41/pelops/issues/72
```

**Comment by agude** [comment]
@d-grossman If you could give this one more look over, I think it's ready to go.

**Comment by agude** [comment]
I have added a patch to work around #76.
